### PR TITLE
Updated Drag Handle

### PR DESCRIPTION
**Pull request by NiclasDev63**
Dragging list items now works as expected and allows dragging multiple nodes.

### CODE REVIEW

**Comment by andrewdoro** [comment]
lgtm

**Comment by andrewdoro** [comment]
We can also reuse your extensions from npm. I find it better when we have composable extensions instead of being handled by library core. (this approach is similar to shadcn-ui components.

Let me know if you think that we can reuse your extensions. Looks like your drag extension can also help with
https://github.com/steven-tey/novel/issues/341

If they can be used, I would recommend the following changes to the PR:
- update package json and update extensions for web demo and example
- change docs to show a drag and drop configuration

Great work💪

**Comment by NiclasDev63** [comment]
I tested locally using my npm extensions instead of integrating the code directly into novel core and everything worked as expected. 

Yes, my extension also solves [#341](https://github.com/steven-tey/novel/issues/341), at least partially, as the scroll speed cannot be configured yet, but auto scrolling works.

I'm just not sure at the moment how the docs should be changed regarding the "show a drag and drop configuration" point. It would be good if you could explain this further.

**Comment by andrewdoro** [comment]
Hey, related to the scroll speed, it's fine for now as long as the user can disable the autoscroll feature. 

Related to docs I had in mind something similar to:
https://github.com/steven-tey/novel/blob/cb682b44a60ae569c0ca70afac9dd9ad8fee3f36/apps/docs/guides/image-upload.mdx

This would have a step by step structure to configure the drag and drop. The content for it can look like this:
1. Install dependecy [tiptap-extension-global-drag-handle](https://github.com/NiclasDev63/tiptap-extension-global-drag-handle)
2. Add Drag extenstion to extension array
3. Anything else that would be helpful for the user, such as the extension config

Let me know wdyt about this. I can also help with the docs.

**Comment by NiclasDev63** [comment]
Thanks, I'll try to add the documentation for the drag handle in the near future.
And as for scrolling, you can disable the autoscroll function within the drag handle by setting scrollTreshold = 0, but you would also have to suppress the default behavior of html to avoid auto scrolling completely, since scrolling is automatically enabled as soon as an element is dragged. This would of course be possible, but I don't think this is an important feature, as auto scrolling is usually expected when dragging elements to the very edge of the screen.

**Comment by andrewdoro** [comment]
Yep that was the thing, just a config for your extension not the html native behavirour.

> And as for scrolling, you can disable the autoscroll function within the drag handle by setting scrollTreshold = 0, 

Could you please update the PR to add your extension for the app web demo and example, and remove the current drag and drop extension from novel. We can add docs later.

**Comment by NiclasDev63** [comment]
I just deleted the drag handle code from novel core and used my npm packages instead. I also added the documentation for both extensions.

**Review by andrewdoro — APPROVED** [review]

File: packages/headless/src/extensions/index.ts [hunk 1]
```diff
@@ -38,7 +40,10 @@ const simpleExtensions = [
     transformCopiedText: true,
   }),
   CustomKeymap,
-  DragAndDrop,
+  GlobalDragHandle.configure({
+    scrollTreshold: 0,
```
Comment (haydenbleasel) [inline packages/headless/src/extensions/index.ts:44]: Typo

**Pull request merged**